sentry-android-core/src/main/java/io/sentry/android/core/performance/TimeSpan.java

```diff
@@ -0,0 +1,164 @@
+package io.sentry.android.core.performance;
+
+import android.os.SystemClock;
+import io.sentry.DateUtils;
+import io.sentry.SentryDate;
+import io.sentry.SentryLongDate;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.jetbrains.annotations.TestOnly;
+
+/**
+ * A measurement for time critical components on a macro (ms) level. Based on {@link
+ * SystemClock#uptimeMillis()} to ensure linear time progression (as opposed to a syncable clock).
+ * To provide real world unix time information, the start uptime time is stored alongside the unix
+ * time. The stop unix time is artificial, it gets projected based on the start time + duration of
+ * the time span.
+ */
+public class TimeSpan implements Comparable<TimeSpan> {
+
+  private @Nullable String description;
+
+  private long startUnixTimeMs;
+  private long startUptimeMs;
+  private long stopUptimeMs;
+
+  /** Start the time span */
+  public void start() {
+    startUptimeMs = SystemClock.uptimeMillis();
+    startUnixTimeMs = System.currentTimeMillis();
+  }
+
+  /**
+   * @param uptimeMs the uptime in ms, provided by {@link SystemClock#uptimeMillis()}
+   */
+  public void setStartedAt(final long uptimeMs) {
+    // TODO maybe sanity check?
+    this.startUptimeMs = uptimeMs;
+
+    final long shiftMs = SystemClock.uptimeMillis() - startUptimeMs;
+    startUnixTimeMs = System.currentTimeMillis() - shiftMs;
+  }
+
+  /** Stops the time span */
+  public void stop() {
+    stopUptimeMs = SystemClock.uptimeMillis();
+  }
+
+  /**
+   * @param uptimeMs the uptime in ms, provided by {@link SystemClock#uptimeMillis()}
+   */
+  public void setStoppedAt(final long uptimeMs) {
+    // TODO maybe sanity check?
+    stopUptimeMs = uptimeMs;
+  }
+
+  public boolean hasStarted() {
+    return startUptimeMs != 0;
+  }
+
+  public boolean hasNotStarted() {
+    return startUptimeMs == 0;
+  }
+
+  public boolean hasStopped() {
+    return stopUptimeMs != 0;
+  }
+
+  public boolean hasNotStopped() {
+    return stopUptimeMs == 0;
+  }
+
+  /**
+   * @return the start timestamp of this measurement, as uptime, in ms
+   */
+  public long getStartUptimeMs() {
+    return startUptimeMs;
+  }
+
+  /**
+   * @return the start timestamp of this measurement, unix time, in ms
+   */
+  public long getStartTimestampMs() {
+    return startUnixTimeMs;
+  }
+
+  /**
+   * @return the start timestamp of this measurement, unix time
+   */
+  public @Nullable SentryDate getStartTimestamp() {
+    if (hasStarted()) {
+      return new SentryLongDate(DateUtils.millisToNanos(getStartTimestampMs()));
+    }
+    return null;
+  }
+
+  /**
+   * @return the start timestamp of this measurement, unix time, in ms
```

Comment:
did you want to say
`   * @return the start timestamp of this measurement, unix time, in seconds`?